proteus/NumericalSolution.py

```diff
@@ -750,7 +751,38 @@ def PUMI2Proteus(self,mesh):
                         nFile=n,
                         analyticalSolution=p.analyticalSolution))
                 model.simTools = self.simOutputList[-1]
-                self.auxiliaryVariables[model.name]= [av.attachModel(model,self.ar[index]) for av in n.auxiliaryVariables]
+                
+                #Code to refresh attached gauges. The goal is to first purge 
+                #existing point gauge node associations as that may have changed
+                #If there is a line gauge, then all the points must be deleted
+                #and remade.
+                from collections import OrderedDict
+                for av in n.auxiliaryVariables:
+                  try:
+                    av.adapted=True
+                    for point, l_d in av.points.iteritems():
+                      if 'nearest_node' in l_d:
+                        l_d.pop('nearest_node')
+                    if(av.isLineGauge or av.isLineIntegralGauge): #if line gauges, need to remove all points
+                      av.points = OrderedDict()
+                    if(av.isGaugeOwner):
+                      for ptGaugeIdx,ptGaugeItem in enumerate(av.pointGaugeVecs):
+                        av.pointGaugeVecs[ptGaugeIdx].destroy();  
+                        av.pointGaugeMats[ptGaugeIdx].destroy();  
+                        av.dofsVecs[ptGaugeIdx].destroy();
+                      av.pointGaugeVecs = []
+                      av.pointGaugeMats = []
+                      av.dofsVecs = []
+                      av.field_ids=[]
+                      av.isGaugeOwner=False
+                  except AttributeError:
+                    pass
```

Comment:
Your summary for what I'm trying to do with the gauges post-adapt sounds right. There is a web of references to the gauge class instance that I didn't want to modify without deeper knowledge of the issue. And since all that should change with the gauges is largely the nearest nodes to perform the computations, I aimed to have that recalculated for pointGauges.